src/EventGridExtension/OutputBinding/EventGridAsyncCollector.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.EventGrid;
+using Microsoft.Azure.EventGrid.Models;
+
+namespace Microsoft.Azure.WebJobs.Extensions.EventGrid
+{
+    public sealed class EventGridAsyncCollector : IAsyncCollector<EventGridEvent>
+    {
+        // use IEventGridClient for mocking test
+        private readonly IEventGridClient _client;
+        private readonly IList<EventGridEvent> _eventsToSend = new List<EventGridEvent>();
+        private readonly string _topicHostname;
+
+        public EventGridAsyncCollector(IEventGridClient client, string topicHostname)
+        {
+            _client = client;
+            _topicHostname = topicHostname;
+        }
+
+        public Task AddAsync(EventGridEvent item, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            _eventsToSend.Add(item);
+
+            return Task.CompletedTask;
+        }
+
+        public async Task FlushAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (_eventsToSend.Any())
+            {
+                await _client.PublishEventsAsync(_topicHostname, _eventsToSend, cancellationToken);
```

Comment:
This function gives me an impression that an `EventGridClient` can send event to multiple topics (since `topicHostName` is an variable/argument). However as far as I know, topic and credentials are tightly bounded, am I right @kalyanaj

Reply:
not sure what you mean... it's not a variable/argument. it's part of the constructor to create the collector which is done by the converter via the attribute; so, it's bound to the parameters of the attribute decorating the collector parameter in the function.

am i misunderstanding your comment?